dapla-manual/blog/posts/2025-06-05-dapla-auth-client/index.qmd

```diff
@@ -0,0 +1,116 @@
+---
+title: "Separering av autentisering i Dapla: Introduksjon av `dapla-auth-client`"
+author: "Johnny Niklasson"
+date: 2025-06-05
+format: html
+categories: [Dapla, Python, Autentisering]
+---
+
+## Innledning
+
+I dette blogginnlegget presenterer vi endringen der autentiseringsklassen er flyttet ut fra `dapla-toolbelt` og inn i en egen Python-pakke kalt `dapla-auth-client`. Hensikten er å gjøre det mulig å installere og bruke autentisering isolert, uten å måtte laste inn hele verktøykassen i `dapla-toolbelt`.
+
+## Hvorfor separere autentisering?
+
+1. **Redusere avhengigheter**  
+   Mange brukere trenger kun autentiseringsbiten for å hente tokens til Google Cloud eller Keycloak. Tidligere måtte man installere og importere hele `dapla-toolbelt`, selv om man bare trengte én liten del. Ved å splitte ut en egen pakke blir installasjonen lettere og mer strømlinjeformet:
+
+   ```toml
+    [tool.poetry.dependencies]
+    dapla-auth-client = ">=0.0.5"
+   ```
```

Comment:
I poetry 2.x har dette blir endret litt:
```diff
- [tool.poetry.dependencies]
- dapla-auth-client = ">=0.0.5"
+ [project]
+ dependencies = [
+    "dapla-auth-client >=3.2.2",
+]
```
Se [denne](https://github.com/statisticsnorway/ssb-project-template-stat/commit/847686b7d5e223a5c4fab9b43c9c7d91571f1b3e) lenken for mer detaljer.